packages/markdown-it-myst/src/directives.ts

```diff
@@ -171,44 +171,44 @@ function directiveArgToTokens(arg: string, lineNumber: number, state: StateCore)
   return nestedPartToTokens('directive_arg', arg, lineNumber, state, 'run_directives', true);
 }
 
-function simplifyDirectiveOptions(options?: [string, string][]) {
+function getDirectiveOptions(options?: [string, string | true][]) {
   if (!options) return undefined;
-  const simplified: Record<string, string | boolean | number> = {};
+  const simplified: Record<string, string | true> = {};
   options.forEach(([key, val]) => {
     if (simplified[key] !== undefined) {
       return;
-    } else if (!isNaN(Number(val))) {
-      simplified[key] = Number(val);
-    } else if (typeof val === 'string' && val.toLowerCase() === 'true') {
-      simplified[key] = true;
-    } else if (typeof val === 'string' && val.toLowerCase() === 'false') {
-      simplified[key] = false;
-    } else {
-      simplified[key] = val;
     }
+    simplified[key] = val;
   });
   return simplified;
 }
 
 function directiveOptionsToTokens(
-  options: [string, string][],
+  options: [string, string | true][],
   lineNumber: number,
   state: StateCore,
 ) {
   const tokens = options.map(([key, value], index) => {
     // lineNumber mapping assumes each option is only one line;
     // not necessarily true for yaml options.
-    const optTokens = nestedPartToTokens(
-      'directive_option',
-      `${value}`,
-      lineNumber + index,
-      state,
-      'run_directives',
-      true,
-    );
+    const optTokens =
+      typeof value === 'string'
+        ? nestedPartToTokens(
+            'directive_option',
+            value,
+            lineNumber + index,
+            state,
+            'run_directives',
+            true,
+          )
+        : [
+            new state.Token('directive_option_open', '', 1),
+            new state.Token('directive_option_close', '', -1),
+          ];
     if (optTokens.length) {
       optTokens[0].info = key;
-      optTokens[0].content = value;
+      optTokens[0].content = typeof value === 'string' ? value : '';
+      optTokens[0].meta = { value };
```

Comment:
I am now putting this in `meta`, which doesn't change the value - content will always change to a string.

Reply:
That makes sense - and we never use `content` either now. We just want `meta.value`